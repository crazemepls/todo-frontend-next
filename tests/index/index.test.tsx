import { render, screen } from '@testing-library/react'
import Home from '../../src/pages/index'
import { useSession } from 'next-auth/react';

jest.mock("next-auth/react", () => {
  const originalModule = jest.requireActual('next-auth/react');
  const mockSession = {
    expires: new Date(Date.now() + 2 * 86400).toISOString(),
    user: { username: "admin" }
  };
  return {
    __esModule: true,
    ...originalModule,
    useSession: jest.fn(() => {
      return {data: mockSession, status: 'authenticated'}  // return type is [] in v3 but changed to {} in v4
    }),
  };
});

jest.mock('next/router', () => ({
  useRouter() {
    return ({
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
      push: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn()
      },
      beforePopState: jest.fn(() => null),
      prefetch: jest.fn(() => null)
    });
  },
}));

describe('Home Render', () => {
  it('should renders the home page', () => {
    const { container } = render(<Home />)

    const createButton = screen.getByText('New Random Todo')

    expect(createButton).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })
})