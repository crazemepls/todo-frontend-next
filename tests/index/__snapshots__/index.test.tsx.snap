// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Render should renders the home page 1`] = `
<div>
  <main
    class="main"
  >
    <div
      style="display: flex; flex-direction: row;"
    >
      <button
        class="mx-5 px-2 bg-gray-500"
      >
        New Random Todo
      </button>
      <button
        class="mx-5 px-2 bg-gray-500"
      >
        New Todo
      </button>
    </div>
    <div
      style="display: flex; flex-direction: row; margin-top: 5px;"
    >
      Sort deadline :
      <button
        class="mx-5 px-2 bg-gray-500"
      >
        Tightest to Loosest
      </button>
    </div>
    <div
      style="display: flex; flex-direction: row;"
    >
      <input
        checked=""
        class="mx-1  ml-8 px-2 bg-gray-500"
        type="checkbox"
      />
      In Progress
      <input
        class="mx-1 ml-8 px-2 bg-gray-500"
        type="checkbox"
      />
      Done
    </div>
    <div
      style="padding-top: 20px;"
    >
      <div
        class="container inline-flex justify-center"
      >
        <ul
          class="inline-flex mt-2"
        >
          <li>
            <a
              class="bg-white border border-gray-300 cursor-pointer text-gray-500 hover:bg-gray-100 hover:text-gray-700 ml-0 rounded-l-lg leading-tight py-2 px-3 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
            >
              Prev
            </a>
          </li>
          <li>
            <a
              class="bg-white border cursor-pointer border-gray-300 text-gray-500 hover:bg-gray-100 hover:text-gray-700 rounded-r-lg leading-tight py-2 px-3 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
            >
              Next
            </a>
          </li>
        </ul>
      </div>
    </div>
  </main>
</div>
`;
